src/components/ik-controller.js

```diff
@@ -87,34 +87,42 @@ AFRAME.registerComponent("ik-controller", {
   update(oldData) {
     if (this.data.leftEye !== oldData.leftEye) {
       this.leftEye = this.el.object3D.getObjectByName(this.data.leftEye);
+      this.leftEye.matrixAutoUpdate = true;
```

Comment:
I can try to do this, but my mental model for `matrixAutoUpdate` if different -- in general I have felt flipping it to true for things where we know there's going to be a tick somewhere probably moving it every frame. i think these joints are a good example, since if they're visible, they're tracked and/or are skinned/IK'd against tracked objects in the skeleton, so ultimately they're probably moving each frame.

i don't mind reconsidering the way we think about when/how to flip the flag. if its OK with you I'd like to do this in later PRs where we try to incrementally remove these. (the benefit here is not expected performance gain, but like you said reducing implicit behavior.) i think probably we should talk through how we want to think about `matrixAutoUpdate` as a team though.

Reply:
as an example of something where `matrixAutoUpdate = true` is set and I'm on the fence about even given my current heuristic of "moves every frame" is the interactable objects, since when pinned/locked we're still computing it. in general, we could probably flip it off, but we'll need to go in and update all the various call sites where the pos/rot/scale are changed like networked-aframe and super hands.

at 30k/ft worth mentioning that a ton of updates are now avoided by a) skipping invisible objects and b) making it a whitelist. so the material performance impact of flipping this flag off in any of the places it's currently on I expect to be minimal.

Reply:
i looked into updating animation mixer, the proper place would be in `PropertyBinding` -- there are a few places however in `SetterByBindingTypeAndVersioning`